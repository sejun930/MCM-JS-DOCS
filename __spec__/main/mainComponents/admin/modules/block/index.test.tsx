import { render, act } from "@testing-library/react";
import { RecoilRoot } from "recoil";

import AdminBlockPage from "src/main/mainComponents/admin/modules/block/block.container.tsx";

let container: HTMLElement;
// 차단 유저 관리 페이지
describe("Admin Block Page", () => {
  // 비동기 함수 처리하는 mock 함수
  jest.mock("../../../../../../src/main/commonsComponents/withAuth", () => ({
    __esModule: true,
    default: jest.fn(() => Promise.resolve(true)), // Resolve with a mock value
  }));

  // window.alert을 사용할 수 있는 mock 함수
  jest.spyOn(window, "alert").mockImplementation(() => {});

  // location의 href 속성 함수로 정의하기
  Object.defineProperty(window, "location", {
    value: { href: "http://example.com" },
    writable: false,
  });

  // location의 reload 속성 함수로 정의하기
  Object.defineProperty(window.location, "reload", {
    writable: true,
    value: jest.fn().mockImplementation(() => {}),
  });

  // 비동기 함수 처리 및 container 초기화
  beforeEach(async () => {
    await act(async () => {
      container = (
        await render(
          <RecoilRoot>
            <AdminBlockPage />
          </RecoilRoot>
        )
      ).container;
    });
  });

  // 스냅샷
  test("Admin Block Page - Snapshot", async () => {
    expect(container).toMatchSnapshot();
  });

  // 유저 리스트 가져오기
  // test("Admin Block Page - Check Has Tags", async () => {
  //   expect(container).toMatchSnapshot();
  // });
});
