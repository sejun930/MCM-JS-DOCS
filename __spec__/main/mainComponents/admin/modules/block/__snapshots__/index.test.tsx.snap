// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Block Page Admin Block Page - Snapshot 1`] = `
<div>
  <div
    class="css-i3mup3"
    id="admin-block-wrapper"
  >
    <div
      class="css-1yxlknd"
    >
      <div
        class="css-ey9ugy"
        id="admin-block-loading"
      >
        <span
          class="mcm-span-unit loading-data css-1oteowz"
        >
          데이터 로딩 중
        </span>
      </div>
      <div
        class="css-qwguda"
        id="admin-block-optional-wrapper"
      >
        <p
          class="mcm-p-unit css-z401r8"
        >
          - 총 
          0
          명의 차단된 유저가 있습니다.
        </p>
        <div
          class="css-haqrkf"
        >
          <div
            class="css-i6f59y"
          >
            <button
              class="mcm-button-unit css-19ngahp"
              role="button_click_event"
              type="submit"
            >
              <img
                class="mcm-image-unit filter-icon css-msby2z"
                src="/images/commons/icons/filter-off.png"
              />
            </button>
          </div>
          <button
            class="mcm-button-unit remove-block-user-btn css-19ngahp"
            role="button_click_event"
            type="submit"
          >
            선택 차단 해제
          </button>
        </div>
      </div>
      <table
        class="css-hcsdpe"
      >
        <thead>
          <tr
            class="css-173yqul"
          >
            <td
              class="block-select"
            />
            <td
              class="block-ip"
            >
              아이피
            </td>
            <td
              class="block-contents"
            >
              차단 사유
            </td>
            <td
              class="block-date"
            >
              차단일
            </td>
            <td
              class="block-cancel"
            >
              차단 해제일
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            class="empty-filter-tr css-173yqul"
          >
            <td
              class="empty-filter-td css-12u6bzw"
            >
              <span
                class="mcm-span-unit css-1oteowz"
              >
                조회된 유저가 없습니다.
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
